R/pmcmc_utils.R

```diff
@@ -1,7 +1,14 @@
 mcstate_pmcmc <- function(pars, probabilities, state, trajectories, restart,
                           predict, chain = NULL, iteration = NULL) {
+
+  if (is_3d_array(pars)) {
+    iteration <- iteration %||% seq.int(0, length.out = nlayer(pars))
+  } else {
+    iteration <- iteration %||% seq.int(0, length.out = nrow(pars))
+  }
```

Comment:
```suggestion
 iteration <- iteration %||% seq.int(0, length.out = last(dim(pars)))
```

This sort of simplification might be useful in the pmcmc_combine

Reply:
`nrow(pars) != last(dim(pars))` though, `ncol(pars)` is right?